### PR TITLE
User create view improvements

### PR DESCRIPTION
**Pull request by TW80000**
### Main
* Hide a bunch of properties when creating a User
* Update client-side validation of a bunch of User properties
  * I used `validateObj` directly instead of `validationPredicates` because that way the back end doesn't do any validation checks. I don't want the back end to check validation because it breaks a large number of tests. I know this is bad, but we don't currently have a good solution for application-specific validation. I plan on thinking about this next, but consider it outside the scope of this PR.
* Made `PasswordView` the default view for `Password` properties
* Disable "Save" button in DAO Controller v2 if data is invalid
* Use properties supplied by `propertyWhitelist` directly so cloned properties with overrides are respected

### Other
* Formatting
* Hid a few props when creating Rules that don't make sense to set, such as `created` and `lastModified`

### CODE REVIEW

File: src/foam/nanos/auth/User.js [hunk 1]
```diff
@@ -184,6 +184,11 @@ foam.CLASS({
       class: 'String',
       name: 'jobTitle',
       section: 'personal',
+      validateObj: function(jobTitle) {
```
Comment (kgrgreer) [inline src/foam/nanos/auth/User.js]: What if we set this property through java or DIG?

File: src/foam/nanos/auth/User.js [hunk 2]
```diff
@@ -373,14 +380,17 @@ foam.CLASS({
...
       width: 100,
       storageTransient: true,
-      validateObj: function (password) {
-        var re = /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9]).{7,32}$/;
-
-        if ( password.length > 0 && ! re.test(password) ) {
-          return 'Password must contain one lowercase letter, one uppercase letter, one digit, and be between 7 and 32 characters in length.';
+      // TODO: Validation that's consistent with the PasswordPolicy feature but
+      // also only applies on create.
+      validateObj: function(desiredPassword, passwordLastModified) {
```
Comment (kgrgreer) [inline src/foam/nanos/auth/User.js]: What about Java/Swift/DIG?

**Comment by TW80000** [comment]
@kgrgreer please review again.

File: src/foam/nanos/auth/email/EmailVerificationWebAgent.java [hunk 3]
```diff
@@ -62,7 +63,10 @@ public void execute(X x) {
       user = (User) user.fclone();
       emailToken.processToken(x, user, token);
     } catch (Throwable t) {
-      message = "Problem verifying your email.<br>" + t.getMessage();
+      String msg = t.getMessage();
+      ((Logger) x.get("logger")).error(msg);
```
Comment (jlhughes) [inline src/foam/nanos/auth/email/EmailVerificationWebAgent.java:67]:
```suggestion
      ((Logger) x.get("logger")).error(msg, t);
```
wil generate a stack trace in the log file.

Comment (TW80000) [reply src/foam/nanos/auth/email/EmailVerificationWebAgent.java:67]: @jlhughes do you want me to revert this then?

**Comment by kgrgreer** [comment]
LGTM

**Pull request merged**